### DISCUSSION TITLE
Understanding Code Base

### DISCUSSION THREAD

**Discussion by ghost** [sButtons/sbuttons#1474] [Q&A] [answered]
Hello,
This is my first contribution. I have cloned the forked repo to my local machine. But I am having a hard time understanding the code base. It would be easier for me to learn and contribute if someone guided me through it.
 Looking forward for proper guidance.
Thank You.

**Comment by shahednasser** [comment, marked as answer]
Hello @nisarga-19 , thank you for your interest to contribute to our project!

I suggest you read the [contributing guidelines](https://github.com/sButtons/sbuttons/blob/2617c1d73a312c350b6b4044b0669f684b4019ef/CONTRIBUTING.md) for a detailed explanation, however, here's a brief overview of the structure of the project:

1. The root project is the project related to the code of the buttons. You'll find the code for the buttons inside `src`, where `_variables.less` includes the variables used for each of the buttons, and the code for the buttons is distributed inside `src/components`.
2. The website's code is under `website` which is a React and Tailwind CSS project. Inside `src` you'll find the React components, and  inside `src/data` you'll find data related to the buttons that are shown on the examples page on the website. So, if you were to add an example of a button, you'd need to add it in `website/src/data/buttons.json`.

As I mentioned, this is a brief overview of the project's structure. I suggest you also read the contributing guidelines to understand better how to contribute.

If you need any other information please let me know.

**Reply by ghost** [reply]
Thank you @shahednasser !!
Did go through the contributing guidelines. I have got a grasp of the project. Will update you if any further queries.
